spec/unit/resources/connection_template/update_spec.rb

```diff
@@ -7,21 +7,39 @@
 
   it 'updates it searching by the Ethernet Network' do
     fake_ethernet = OneviewSDK::EthernetNetwork.new(@client, connectionTemplateUri: 'fake/connection-template')
+    fake_fc = OneviewSDK::FCNetwork.new(@client, connectionTemplateUri: 'fake/connection-template')
+    fake_fcoe = OneviewSDK::FCoENetwork.new(@client, connectionTemplateUri: 'fake/connection-template')
+    fake_net = OneviewSDK::NetworkSet.new(@client, connectionTemplateUri: 'fake/connection-template')
     allow(OneviewSDK::EthernetNetwork).to receive(:find_by).and_return([fake_ethernet])
+    allow(OneviewSDK::FCNetwork).to receive(:find_by).and_return([fake_fc])
+    allow(OneviewSDK::FCoENetwork).to receive(:find_by).and_return([fake_fcoe])
+    allow(OneviewSDK::NetworkSet).to receive(:find_by).and_return([fake_net])
     allow_any_instance_of(OneviewSDK::ConnectionTemplate).to receive(:exists?).and_return(true)
     allow_any_instance_of(OneviewSDK::ConnectionTemplate).to receive(:retrieve!).and_return(true)
     allow_any_instance_of(OneviewSDK::ConnectionTemplate).to receive(:like?).and_return(false)
-    expect_any_instance_of(OneviewSDK::ConnectionTemplate).to receive(:update).and_return(true)
+    allow_any_instance_of(OneviewSDK::ConnectionTemplate).to receive(:update).and_return(true)
```

Comment:
Awesome idea, was bothered by the allow as well but couldn't fit an expect there due to multiple resource instances.
Updating the PR atm.